src/server/img_storage.rs

```diff
@@ -214,11 +237,17 @@ impl StoreImage {
     /// We don't do any form of check to see if it matches what we got before.
     /// If you have "Storage Legacy Bucket Writer" previous content is overwritten.
     /// (e.g. set the path to be the SHA1 of the bytes or whatever.)
-
     pub async fn store(&self, uri: &uri::Uri) -> HandlerResult<StoredImage> {
+        if let Some(image) = self.fetched_images.get(uri) {
+            if !image.expired() {
+                return Ok(image.clone());
+            }
+        }
         let (image, content_type) = self.fetch(uri).await?;
         let metrics = self.validate(uri, &image, &content_type).await?;
-        self.upload(image, &content_type, metrics).await
+        let image = self.upload(image, &content_type, metrics).await?;
```

Comment:
nit: can we call this `stored_image` or something less confusing with the `image:Bytes` we had before?